### ISSUE TITLE
pyparsing missing from install

### ISSUE THREAD

**Issue by micah** [searx/searx#887]
If I follow the installation instructions (on Debian Jessie), on the ```./manage.sh update_packages``` step, the following happens:

```
...
  Found existing installation: setuptools 5.5.1
    Uninstalling setuptools:
      Successfully uninstalled setuptools
  Running setup.py install for cffi
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/setuptools/__init__.py", line 12, in <module>
        import setuptools.version
      File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/setuptools/version.py", line 1, in <module>
        import pkg_resources
      File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 72, in <module>
        import packaging.requirements
      File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/packaging/requirements.py", line 9, in <module>
        from pyparsing import stringStart, stringEnd, originalTextFor, ParseException
    ImportError: No module named pyparsing
    Complete output from command /srv/searx/searx-ve/bin/python2 -c "import setuptools, tokenize;__file__='/tmp/pip-build-DzEoyT/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-HZc3xx-record/install-record.txt --single-version-externally-managed --compile --install-headers /srv/searx/searx-ve/include/site/python2.7:
    Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/setuptools/__init__.py", line 12, in <module>

    import setuptools.version

  File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/setuptools/version.py", line 1, in <module>

    import pkg_resources

  File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 72, in <module>

    import packaging.requirements

  File "/srv/searx/searx-ve/local/lib/python2.7/site-packages/packaging/requirements.py", line 9, in <module>

    from pyparsing import stringStart, stringEnd, originalTextFor, ParseException

ImportError: No module named pyparsing

----------------------------------------
Cleaning up...
Command /srv/searx/searx-ve/bin/python2 -c "import setuptools, tokenize;__file__='/tmp/pip-build-DzEoyT/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-HZc3xx-record/install-record.txt --single-version-externally-managed --compile --install-headers /srv/searx/searx-ve/include/site/python2.7 failed with error code 1 in /tmp/pip-build-DzEoyT/cffi
Storing debug log for failure in /srv/searx/.pip/pip.log
```

This is easily fixed by doing this:
```
(searx-ve)searx@searx:~$ pip install pyparsing
Downloading/unpacking pyparsing
  Downloading pyparsing-2.2.0-py2.py3-none-any.whl (56kB): 56kB downloaded
Installing collected packages: pyparsing
Successfully installed pyparsing
Cleaning up...
(searx-ve)searx@searx:~$ ./manage.sh update_packages
Requirement already satisfied (use --upgrade to upgrade): certifi==2017.1.23 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 1))
Requirement already satisfied (use --upgrade to upgrade): flask==0.12 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): flask-babel==0.11.1 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 3))
Requirement already satisfied (use --upgrade to upgrade): lxml==3.7.3 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 4))
Requirement already satisfied (use --upgrade to upgrade): idna==2.5 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 5))
Requirement already satisfied (use --upgrade to upgrade): pygments==2.1.3 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 6))
Requirement already satisfied (use --upgrade to upgrade): pyopenssl==16.2.0 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 7))
Requirement already satisfied (use --upgrade to upgrade): python-dateutil==2.6.0 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 8))
Requirement already satisfied (use --upgrade to upgrade): pyyaml==3.12 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 9))
Requirement already satisfied (use --upgrade to upgrade): requests[socks]==2.13.0 in ./searx-ve/lib/python2.7/site-packages (from -r /srv/searx/requirements.txt (line 10))
  Installing extra requirements: 'socks'
Requirement already satisfied (use --upgrade to upgrade): Babel>=2.3 in ./searx-ve/lib/python2.7/site-packages (from flask-babel==0.11.1->-r /srv/searx/requirements.txt (line 3))
Requirement already satisfied (use --upgrade to upgrade): Jinja2>=2.5 in ./searx-ve/lib/python2.7/site-packages (from flask-babel==0.11.1->-r /srv/searx/requirements.txt (line 3))
Cleaning up...
(searx-ve)searx@searx:~$ 
```

But it seems the script needs updating to include that.

**Comment by Vertux** [comment]
I can confirm this issue.

**Comment by dalf** [comment]
I can reproduce the bug with a Debian Jessie not a Debian Wheezy (using a docker image).
Installing the [python-pyparsing](https://packages.debian.org/fr/jessie/python-pyparsing) debian package doesn't help.

**Comment by asciimoo** [comment]
I think this is not a searx specific bug. Pyparsing is not a direct dependency of searx. The root cause is that pip and setuptools are outdated.
`pip install --upgrade pip && pip install --upgrade setuptools` solves the problem.

**Comment by micah** [comment]
Adam Tauber <notifications@github.com> writes:

> I think this is not a searx specific bug. Pyparsing is not a direct dependency of searx. The root cause is that pip and setuptools are outdated.
> `pip install --upgrade pip && pip install --upgrade setuptools` solves the problem.

probably, but searx could make this smoother for people who are in this
situation, no?

**Comment by asciimoo** [comment]
> probably, but searx could make this smoother for people who are in this situation, no?

Agree, `manage.sh` could check pip and setuptools version before package updates.

**Comment by dalf** [comment]
Fixed with PR #972
